pkg/agent/flowexporter/connections/connections.go

```diff
@@ -234,26 +240,36 @@ func (cs *ConnectionStore) Poll() (int, error) {
 	// We do not expect any error as resetConn is not returning any error
 	cs.ForAllConnectionsDo(resetConn)
 
-	filteredConnsList, totalConns, err := cs.connDumper.DumpFlows(openflow.CtZone)
-	if err != nil {
-		return 0, err
+	var zones []uint16
+	var connsLens []int
+	if cs.v4Enabled {
+		zones = append(zones, openflow.CtZone)
 	}
-	// Update only the Connection store. IPFIX records are generated based on Connection store.
-	for _, conn := range filteredConnsList {
-		cs.addOrUpdateConn(conn)
+	if cs.v6Enabled {
+		zones = append(zones, openflow.CtZoneV6)
 	}
-	connsLen := len(filteredConnsList)
-	filteredConnsList = nil
-	metrics.TotalConnectionsInConnTrackTable.Set(float64(totalConns))
-	maxConns, err := cs.connDumper.GetMaxConnections()
-	if err != nil {
-		return 0, err
-	}
-	metrics.MaxConnectionsInConnTrackTable.Set(float64(maxConns))
+	for _, zone := range zones {
+		filteredConnsList, totalConns, err := cs.connDumper.DumpFlows(zone)
+		if err != nil {
+			return []int{}, err
+		}
+		// Update only the Connection store. IPFIX records are generated based on Connection store.
+		for _, conn := range filteredConnsList {
+			cs.addOrUpdateConn(conn)
+		}
+		connsLens = append(connsLens, len(filteredConnsList))
+		filteredConnsList = nil
```

Comment:
Not your PR. This can be removed. Thanks.

Reply:
Updated.